### PR TITLE
fix visualization for siamese/merge models

### PR DESCRIPTION
**Pull request by temporaer**
Without this, recursion of `_process_layer` stopped at `Siamese` layers. Also, since `Siamese` layers result in a model which is not a tree, we have to keep track of already seen nodes so we don't trace them twice (introducing additional edges)

### CODE REVIEW

File: keras/utils/visualize_util.py [hunk 1]
```diff
@@ -14,7 +14,8 @@
 
 
 def layer_typename(layer):
-    return type(layer).__module__ + "." + type(layer).__name__
+    # return type(layer).__module__ + "." + type(layer).__name__
```
Comment (fchollet) [inline keras/utils/visualize_util.py]: What's up with this line?

Comment (temporaer) [inline keras/utils/visualize_util.py]: Oh, sorry. That got mixed in accidentally. I'll remove it tomorrow morning!

**Comment by temporaer** [comment]
@fchollet fixed.

**Comment by fchollet** [comment]
Ok, looks good to me.

**Comment by temporaer** [comment]
@fchollet did you want to merge this one? Your message sounds like you did ;)

**Comment by temporaer** [comment]
@fchollet your message reads as if you'd like to merge this commit, but you closed the PR without merging. Was this a mistake?

**Pull request closed without merging**